operators/src/plot/temporal_raster_mean_plot.rs

```diff
@@ -0,0 +1,438 @@
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedPlotOperator,
+    Operator, PlotOperator, PlotQueryProcessor, PlotResultDescriptor, QueryContext, QueryProcessor,
+    QueryRectangle, RasterQueryProcessor, TypedPlotQueryProcessor,
+};
+use crate::error;
+use crate::util::math::average_floor;
+use crate::util::Result;
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use geoengine_datatypes::plots::{AreaLineChart, Plot, PlotData};
+use geoengine_datatypes::primitives::{Measurement, TimeInstance, TimeInterval};
+use geoengine_datatypes::raster::{Pixel, RasterTile2D};
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+use std::collections::BTreeMap;
+
+pub const TEMPORAL_RASTER_MEAN_PLOT_NAME: &str = "Temporal Area Mean Plot";
+
+/// A plot that shows the mean values of rasters over time as an area plot.
```

Comment:
Why stop at mean btw? Do we want to create separate operators for min/max/median over time? Maybe it would be nice to have a StatisticsOverTime plot where users can choose from various statistics, and the selected ones are all plotted into a line chart? This is of course beyond the scope of this PR but a general decision we have to make. Lots of plots, or few rich ones.

Reply:
You have several options, also to compare multiple datasets in one plot. I would say we extend the operator based on demands :shrug:. In the end we want to be able to combine stuff, but I guess we can't do it prematurely without use case because of other functionality.

Reply:
True, but we also have to think about the future proofness of the operators. Once there are workflows using a plot we have to keep it around forever. And we can't just add min/max values to a plot named "Mean Plot". Then we'd have at some point a general StatisticsOverTime and a special Mean plot.

Maybe this is a separate task "Polish API" for a 1.0 release though and until then we don't care too much about it?

Reply:
Yeah, I think we aren't there yet. We could also deprecate operators and switch all occurrences of the old one with a spec of the new one.